client/jetpack-cloud/sections/partner-portal/sidebar/index.tsx

```diff
@@ -11,105 +11,100 @@ import { itemLinkMatches } from 'calypso/my-sites/sidebar/utils';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import 'calypso/components/jetpack/sidebar/style.scss';
 
-interface Props {
+type Props = {
 	path: string;
-	dispatchRecordTracksEvent: typeof recordTracksEvent;
-	translate: typeof TranslateType;
-}
-class PartnerPortalSidebar extends Component< Props > {
-	onNavigate = ( menuItem: string ) => () => {
-		this.props.dispatchRecordTracksEvent( 'calypso_jetpack_sidebar_menu_click', {
-			menu_item: menuItem,
-		} );
+};
 
-		window.scrollTo( 0, 0 );
-	};
+const PartnerPortalSidebar: React.FC< Props > = ( { path } ) => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
 
-	render() {
-		const { translate, path } = this.props;
+	const onNavigate = useCallback(
+		( menuItem: string ) => () => {
+			dispatch(
+				recordTracksEvent( 'calypso_jetpack_sidebar_menu_click', {
+					menu_item: menuItem,
+				} )
+			);
+		},
+		[ dispatch ]
+	);
 
-		return (
-			<Sidebar className="sidebar__jetpack-cloud">
-				<SidebarRegion>
-					<SidebarMenu>
+	const arePaymentsEnabled = isEnabled( 'jetpack/partner-portal-payment' );
```

Comment:
It looks like the `jetpack/partner-portal-payment` is enabled on all environments, so we can probably delete it. 

As it's out of the scope of this task, I added a card to maintenance: 1202619025189113-as-1205140782121819